### ISSUE TITLE
Gem hangs on install, looking for non-existent Ruby 2.4.0 content

### ISSUE THREAD

**Issue by JakeTheSnake3p0** [phusion/passenger#1926]
This is using Windows 7 x64 host with Vagrant running Ubuntu 16.04 in VirtualBox. RVM using Ruby 2.4.0 and the global RVM gemset.

I cannot install the passenger gem as the installation hangs in the process. I ran it using `gem install passenger -V` and it stops at the following output:

> Attempting to download https://oss-binaries.phusionpassenger.com/binaries/passenger/by_release/5.1.2/rubyext-ruby-2.4.0-x86_64-linux.tar.gz into /home/ubuntu/.rvm/rubies/ruby-2.4.0/lib/ruby/gems/2.4.0/gems/passenger-5.1.2/download_cache

I then copied the url and browsed to [the index](https://oss-binaries.phusionpassenger.com/binaries/passenger/by_release/5.1.2/), that archive is nowhere to be found. This probably causes a 404 that isn't handled well.

**Comment by OnixGH** [comment]
The rubyext are optional and we don't supply them for every possible Ruby version. It is correct that this archive is not there.

So a 404 shouldn't stall the process (for me gem install works fine on 16.04 / 2.4.0 with the 404), I'd say your problem is likely something other than Passenger.

**Comment by JakeTheSnake3p0** [comment]
It was another issue that was tangentially related. `apt-get update` would hang on a certain URL indefinitely until I found [this page](http://askubuntu.com/a/769931/248266); I compared that source list with my own (vagrant xenial x64), uncommented the partner sources in my list, then ran `apt-get update` again and it finally failed the installation of the hanging URL instead of just hanging. I then re-commented the two partner sources and then ran the gem installation and it worked.
